src/backend/joanie/tests/core/api/activity_log/test_create.py

```diff
@@ -0,0 +1,80 @@
+"""
+Test suite for ActivityLog create API endpoint.
+"""
+
+from http import HTTPStatus
+from unittest.mock import patch
+from uuid import uuid4
+
+from django.test import override_settings
+
+from joanie.core import enums, factories, models
+from joanie.tests.base import BaseAPITestCase
+
+
+@override_settings(JOANIE_ACTIVITY_LOG_SECRETS=["shared secret"])
+class ActivityLogCreateApiTest(BaseAPITestCase):
+    """
+    Test suite for ActivityLog create API endpoint.
+    """
+
+    def test_api_activity_log_create_anonymous(self):
+        """
+        Anonymous users should not be able to get activity logs
+        """
+        data = {
+            "type": enums.ACTIVITY_LOG_TYPE_PAYMENT_SUCCEEDED,
+            "level": enums.ACTIVITY_LOG_LEVEL_INFO,
+        }
+        response = self.client.post(
+            "/api/v1.0/activity-logs/", data=data, content_type="application/json"
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.FORBIDDEN)
+        self.assertEqual(models.ActivityLog.objects.count(), 0)
+
+    def test_api_activity_log_create_user(self):
+        """
+        User should not be able to create activity logs
+        """
+        user = factories.UserFactory()
+        jwt_token = self.generate_token_from_user(user)
+        data = {
+            "type": enums.ACTIVITY_LOG_TYPE_PAYMENT_SUCCEEDED,
+            "level": enums.ACTIVITY_LOG_LEVEL_INFO,
+        }
+
+        response = self.client.post(
+            "/api/v1.0/activity-logs/",
+            data=data,
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.UNAUTHORIZED)
+        self.assertEqual(models.ActivityLog.objects.count(), 0)
+
+    @patch("joanie.core.api.client.check_signature")
```

Comment:
Instead of mocking it, why not generating a good signature by not generating randomly in your test the user id and the order id ?

Reply:
IMO, mocking reduces noise in the test, and increases maintainability.
Besides, we don't want to test `check_signature` here.
But, we can remove the mock if you want.

Reply:
ok